src/chainlit/server.py

```diff
@@ -308,7 +310,15 @@ async def serve(path: str):
                 if os.path.isfile(file_path):
                     return FileResponse(file_path)
 
-        return HTMLResponse(content=html_template, status_code=200)
+        html_template = get_html_template()
+
+        response = HTMLResponse(content=html_template, status_code=200)
+
+        response.set_cookie(
+            key="chainlit-headers",
+            value=json.dumps(dict(request.headers)),
```

Comment:
You might want to switch to HttpOnly cookies. This means the cookies won't be able to be read by the frontend code at all.

```suggestion
            value=json.dumps(dict(request.headers)),
            httponly=True,
```

This is compatible with the current way you are getting the cookie in the socket.io `connect` handler.

Reply:
Done! Also added a test!